Source/scrollrt.cpp

```diff
@@ -916,6 +918,139 @@ void DrawDungeon(const Surface &out, Point tilePosition, Point targetBufferPosit
 	}
 }
 
+Point CheckPredefinedPattern(Point tilePosition)
+{
+	Point pos = { 0, 0 };
+	if (tilePosition.y == -1 && tilePosition.x >= 52 && tilePosition.x <= 61) {
+		// top side sand patch
+		if (tilePosition.x == 61) {
+			// right corner
+			pos = { 64, 66 };
+		} else if (tilePosition.x == 52) {
+			// left corner
+			pos = { 57, 66 };
+		} else {
+			// edge
+			pos = { 61, 66 };
+		}
+	} else if (tilePosition.x == 96 && ((tilePosition.y >= 52 && tilePosition.y <= 71) || (tilePosition.y >= 79 && tilePosition.y <= 89))) {
+		// right side sand patch
+		if (tilePosition.y == 89 || tilePosition.y == 71) {
+			// bottom corner
+			pos = { 65, 74 };
+		} else if (tilePosition.y == 79 || tilePosition.y == 52) {
+			// top corner
+			pos = { 65, 67 };
+		} else {
+			// edge
+			pos = { 65, 72 };
+		}
+	} else if ((tilePosition.y < 0 && IsAnyOf(tilePosition.x, 2, 62, 82))
+	    || (tilePosition.y > 95 && IsAnyOf(tilePosition.x, 32, 84))) {
+		// left river bank
+		pos = { 68, 22 - abs(tilePosition.y % 3) };
+	} else if ((tilePosition.y < 0 && IsAnyOf(tilePosition.x, 3, 63, 83))
+	    || (tilePosition.y > 95 && IsAnyOf(tilePosition.x, 33, 85))) {
+		// right river bank
+		pos = { 69, 22 - abs(tilePosition.y % 3) };
+	}
+	return pos;
+}
+
+Point FillPattern(Point tilePosition)
+{
+	constexpr int PATTERNWIDTH = 5;
+	constexpr int PATTERNHEIGHT = 5;
+	constexpr std::array<Point, 6> patterns { { { 51, 27 }, { 23, 57 }, { 54, 34 }, { 79, 46 }, { 41, 32 }, { 52, 21 } } };
+
+	static int patternRandomizer[MAXDUNX][MAXDUNY] {};
+	static bool doInit = true;
+
+	if (doInit) {
+		doInit = false;
+		std::mt19937 BetterRng;
+		BetterRng.seed(666);

```

Comment:
Using a constant seed means the same pattern will be generated each run of the game, previously this would change each time the game was launched?

If you wanted it to be random on load can use `std::random_device` to init the engine
```suggestion
		std::random_device rd;
		std::mt19937 BetterRng(rd());
```

Reply:
I wanted it to be the same